gitlab-lib.yaml

```diff
@@ -57,6 +57,68 @@
         fi
       done
 
+.pnnllib-gitlab-mirror-oci-artifacts:
+    # Arguments are passed in environment variables
+    # ARTIFACTS - Newline separated list of oci artifacts to mirror
+    # ARTIFACTS_FILE - Optional artifact filename which lists the desired oci artifacts to be cached in a multiline format
+    # PREFIX - extra prefix for the artifacts. Can be used to add a string on the front of an image name or a parent directory in the repository.
+    # RETAG - Change the tag or sha of the image. to rename the image from foo:bar to foo:baz, set to "\1:baz"
+    # SOURCE_REGISTRY_USERNAME - Optional username to use for the source registry
+    # SOURCE_REGISTRY_PASSWORD - Optional password to use for the source registry
+    # DEST_REGISTRY - Optional custom destination registry, defaults to CI_REGISTRY
+    # DEST_REGISTRY_PATH - Optional custom destination registry path, defaults to CI_PROJECT_PATH
+    # DEST_REGISTRY_USERNAME - Optional username to use for destination registry, defaults to CI_REGISTRY_USER
+    # DEST_REGSITRY_PASSWORD - Optional password to use for destination registry, defaults to CI_REGISTRY_PASSWORD
+    # OVERWRITE - Set this to anything to allow overwriting artifacts that already exist
+    # COPY_LOCAL - Set this to anything to copy artifacts to local disk before pushing to the destination,
+    #   useful when pulling and pushing to the same registry with separate credentials
+    image: quay.io/skopeo/stable:latest
```

Comment:
Does latest actually have a shell now? That was a problem in the past.

We may want to consider pinning it too. If they change behavior of the container, things probably shouldn't break.

Reply:
I've pinned the image now.  The v1.12.0 image have /bin/sh in it and is built from a Fedora container image variant based on the /etc/os-release output.